src/main/java/com/clevertap/maven/plugins/supertest/SuperTestMavenPlugin.java

```diff
@@ -150,7 +150,12 @@ public String createRerunCommand(Map<String, List<String>> classnameToTestcaseLi
         for (String className : classnameToTestcaseList.keySet()) {
             List<String> failedTestCaseList = classnameToTestcaseList.get(className);
             if (!failedTestCaseList.isEmpty()) {
-                retryRun.append(className).append("#");
+                retryRun.append(className);
+                if(failedTestCaseList.contains("")) {
```

Comment:
What if it it contains `"" and "testName"`? Should the condition be updated? To ignore empty entries in the list?

Reply:
if it contains "" and "testName", we rerun the entire class. We don't need to specify the tests to run here. We rerun the entire class.

Reply:
the classXML1 string actually demonstrates this behaviour.